pyrex.py

```diff
@@ -317,11 +317,13 @@ def get_engine_info(config):
 
 def get_subid_length(filename, name):
     with open(filename, "r") as f:
-        for line in f:
-            (ident, _, id_length) = line.rstrip().split(":")
-            if ident == name:
-                return int(id_length)
-    return 0
+        content = '\n'.join(f.readlines())
```

Comment:
I think I'd rather see:
```
  with open(filename, "r") as f:
      r = re.compile(r'{}:\d+:(\d+)'.format(re.escape(name)))
      for line in f:
          m = r.match(line)
          if m is not None:
              return int(m.group(1))
  return 0
```

Reply:
Could you explain why `name` is now escaped in your suggestion? Is there any technical reason for this?

Reply:
It's probably unlikely that `name` contains a special regex character, but I have the habit of escaping any strings I'm putting in a regex that are supposed to be taken literally.